src/datahike/query.cljc

```diff
@@ -539,11 +542,44 @@
      :clj  (when (namespace sym)
              (when-some [v (resolve sym)] @v))))
 
+(def ^:private find-method
+  #?(:cljs nil
+     :clj (memoize
+           (fn find-method-impl [^Class this-class method-name args-classes]
+             (or (->> this-class
+                      .getMethods
+                      (some (fn [^Method method]
+                              (when (and (= method-name (.getName method))
+                                         (= (count args-classes)
+                                            (.getParameterCount method))
+                                         (every? true? (map #(Reflector/paramArgTypeMatch %1 %2)
+                                                            (.getParameterTypes method)
+                                                            args-classes)))
+                                method))))
+                 (throw (ex-info (str (.getName this-class) "."
+                                      method-name "("
+                                      (str/join "," (map #(.getName ^Class %) args-classes))
+                                      ") not found")
+                                 {:this-class this-class
+                                  :method-name method-name
+                                  :args-classes args-classes})))))))
+
+
+(defn- resolve-method [method-sym]
+  #?(:cljs nil
+     :clj (let [method-str (name method-sym)]
+            (when (= \. (.charAt method-str 0))
+              (let [method-name (subs method-str 1)]
+                (fn [this & args]
+                  (let [^Method method (find-method (class this) method-name (mapv class args))]
+                    (Reflector/prepRet (.getReturnType method) (.invoke method this (into-array Object args))))))))))
```

Comment:
Note: `Reflector/prepRet` is an important part of interop because Clojure recognizes only `Boolean/FALSE` and `nil` as falsey, and not any other Boolean instance like `(Boolean. false)`, for example `(if (Boolean. false) :then :else)` returns `:then`. Java method invocations create new Booleans instead of reusing existing `Boolean/TRUE` and `Boolean/FALSE`, and `prepRet` ensures that boolean method invocation results are either `Boolean/TRUE` or `Boolean/FALSE`

Reply:
Thanks @vlaaad. I'll have a look today and come back to you.